lib/package_config.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// A package configuration is a way to assign file paths to package URIs,
+/// and vice-versa,
+library package_config.package_config;
+
+import "dart:io" show File, Directory;
+import "src/discovery.dart" as discover;
+import "src/package_config.dart";
+import "src/package_config_json.dart";
+
+export "src/package_config.dart" show PackageConfig, Package;
+
+/// Reads a specific package configuration file.
+///
+/// The file must exist and be readable.
+/// It must be either a valid `package_config.json` file
+/// or a valid `.packages` file.
+/// It is considered a `package_config.json` file if its first character
+/// is a `{`.
+///
+/// If the file is a `.packages` file, first checks if there is a
+/// `.dart_tool/package_config.json` file next to the original file,
+/// and if so, loads that instead.
+///
+/// If [extraData] is provided, and the loaded file is a `package_config.json`
```

Comment:
What is this referencing?

Reply:
Good catch. That is now put directly on the `PackageConfig` class.